src/sumo_backend/compare.clj

```diff
@@ -0,0 +1,335 @@
+(ns sumo-backend.compare)
+(require '[sumo-backend.mysql :as db])
+(require '[sumo-backend.utils :as utils])
+
+;; Note on this namespace:
+;; Similar to how get-rikishi-losses-to-opponent
+;; and get-rikishi-wins-against-opponent
+;; take a more optimized list of bouts to parse through,
+;; see if other functions here can take a more
+;; optimized list of bouts to parse through.
+;; currently, most of these functions parse through
+;; every bout, looking for their success-criteria.
+;; that will get slow as more data is added.
+
+;; Note on ranks
+;; Komusubi, Maegashira, and Juryo are not the same
+;; Difference between them is the Maegashira's number
+;; Or all Maegashira numbers + the Juryo number
+;; e.g. Maegashira #1 is one rank away from Komusubi
+
```

Comment:
I think that the rank comparision could be simplified by enumerating the possible ranks, like `yokozuna=1` etc here, but drop the rank-and-file number and just keep counting up, something like `maegashira_1=5`, `maegashira_2=6`.  Internally (in the DB and in the code), we would only care about the rank number, which would make it easy to query against ranks in the DB, sort by ranks using simple `sort-by` calls, etc.  Converting between the human-readable rank str and the machine-readable rank int would be handled by `sumo-backend.compare/rank-to-label` and `sumo-backend.compare/label-to-rank`

Reply:
Didn't realize that there were different numbers of Maegashira each tournament, which could complicate this.

Reply:
I've reworked this based on part of what you said. It's better than what I had to start with.

I went with having a function that returns a map of rank -> value per tournament.
All these rank comparisons are on bouts that happened in tournaments, so something like a 5 rank level upset is a 5 level rank upset, regardless of if it's:
- Juryo #1 -> Maegashira #13 (in the case of having 17 Maegashira in an arbitrary tournament)
- or -
- Juryo #1 -> Maegashira #12 (in the case of having 16 Maegashira in an arbitrary tournament)

or any other situation like that too. So, your idea, but contextualized per tournament. 
And I wanted to wrap each one of those function calls to generate that rank -> value per tournament in `memoize` but was having an issue with that I wanted to ask about